### PR TITLE
don't avoid walking on own sites

### PR DESCRIPTION
**Pull request by karlthepagan**
allows walking on own road and rampart sites

### CODE REVIEW

File: room.js [hunk 1]
```diff
@@ -857,7 +857,7 @@ mod.extend = function(){
                         let setCosts = structure => {
                             const site = structure instanceof ConstructionSite;
                             // don't walk on allied construction sites.
-                            if (site && Task.reputation.allyOwner(structure)) return costMatrix.set(structure.pos.x, structure.pos.y, 0xFF);
+                            if (site && !structure.my && Task.reputation.allyOwner(structure)) return costMatrix.set(structure.pos.x, structure.pos.y, 0xFF);
```
Comment (Spedwards) [inline room.js]: Doesn't `allyOwner` include yourself? If so this will never pass.

Comment (karlthepagan) [inline room.js]: My site: true && false && true
Ally site: true && true && true

Comment (karlthepagan) [inline room.js]: Tested here, https://screeps.com/a/#!/room/W29N17

```
[7:29:14 PM]TRAVELER: incomplete path for miner-650-1 from [room W29N17 pos 36,30] to [room W29N17 pos 39,41]. Route length 0.
[7:29:22 PM]TRAVELER: incomplete path for hauler-500-1 from [room W29N17 pos 36,30] to [room W29N17 pos 39,41]. Route length 0.
[7:29:26 PM]TRAVELER: incomplete path for hauler-1100-1 from [room W29N17 pos 42,28] to [room W29N17 pos 39,26]. Route length 0.
[7:29:30 PM]TRAVELER: incomplete path for upgrader-950-1 from [room W29N17 pos 32,14] to [room W29N17 pos 30,9]. Route length 0.
[7:29:30 PM]TRAVELER: incomplete path for miner-650-1 from [room W29N17 pos 35,30] to [room W29N17 pos 39,41]. Route length 0.
[7:29:38 PM]TRAVELER: incomplete path for hauler-1100-1 from [room W29N17 pos 42,29] to [room W29N17 pos 39,26]. Route length 0.
[7:29:38 PM]TRAVELER: incomplete path for hauler-500-1 from [room W29N17 pos 35,30] to [room W29N17 pos 39,41]. Route length 0.
[7:29:42 PM]TRAVELER: incomplete path for miner-650-1 from [room W29N17 pos 36,30] to [room W29N17 pos 39,41]. Route length 0.
 delete Memory.pathfinder.W29N17
 true
Game.rooms.W29N17.structureMatrix.get(36,22)
[7:29:50 PM] W29N17 > Calulating cost matrix
1
```

Comment (Spedwards) [inline room.js]: Oh, right. You want it to not pass.

**Comment by karlthepagan** [comment]
Also just to be sure I tested that all of my sites were my truthy, including roads and walls.

**Pull request merged**